src/terraingenerator/main.cpp

```diff
@@ -73,28 +89,132 @@ static void showVersion()
                << qPrintable(QCoreApplication::applicationVersion());
 }
 
-static void parseCommandLineArguments(CommandLineOptions &options)
+static bool parseCommandLineArguments(CommandLineOptions &options)
 {
     const QStringList arguments = QCoreApplication::arguments();
+    bool inPriority = false;
+    bool inSource = false;
+    bool inCombine = false;
+    int argCount = 0;
+    QStringList combine;
```

Comment:
Instead of having this combine variable, you could just append an empty `QStringList` to `options.combineList` and use `options.combineList.last().append(...)` to add to it. That avoids the need to add it later as well as to clear it.